### ISSUE TITLE
Wrong Weekly note name using ISO week format GGGG-[W]WW

### ISSUE THREAD

**Issue by linhtto** [liamcain/obsidian-periodic-notes#128]
- I choose "GGGG-[W]WW" as the name of the weekly note in the plugin setting. It shows me that today should be 2022-W27.
- However, if I choose to create a weekly note today it'll show 2022-W26. 
- Furthermore, ISO week starts on a Monday, and my computer is set as such as well, but the note name changes on a Sunday instead.

**Comment by EvanEdwards** [comment]
<deleted misleading comment after more research, and realizing that moment.js was being used>

It is displaying correctly.  You (and briefly myself) are confused: there are two different week numbering systems, and they are often off by one.  ISO (international standard) uses WW.  The other (which I don't use myself, but some US businesses do) uses ww (lowercase).  You would want to use YYYY-[W]ww for your non-ISO week number.  For ISO numbering, GGGG-[W]WW is correct.

HOWEVER -- it is misleading because the Calendar Obsidian plugin is using non-ISO week numbers.  I was confused for a bit until I checked it against GNU date and KDE's calendar.  Today (July 7th or 7 July) is week 27.  In the Calendar plugin it is listed as week 28.  There is now a Locale option in Calendar, and if you set it to the US, this incorrect behavior exists (people seem to think we don't use international standards here in Nashville, but we do!).  If you set it to, say, Italy, you get Italian spelling, but the week numbers are correct.

The two plugins disagree about what the week number should be, but at this point I believe Periodic Notes is correct and their documentation is correct.

**Comment by zhaohongxuan** [comment]
Same issue on IOS, but it works correctly in macOS and android, seems a bug in obsidian library `window.moment`

**Comment by nmariano9** [comment]
Same issue here, although I'm on macOS. I do have the Calendar plugin installed. Any fix for this yet?

**Comment by Smitty010** [comment]
A "solution" I found on the Calendar plugin github issues is to set your locale to "en-gb" (even if you are in the US).  The calendar plugin was not showing the correct week numbers.  Week 1 in 2022 should start January 3.  As I understand it, week 1 is the first week in the calendar year that contains a Thursday (https://en.wikipedia.org/wiki/ISO_week_date).  Not sure if the problem is Calendar or Momentjs (I'm suspicious its the latter), but things seemed to work as I expected once I changed the locale.  I glanced at several years into the future and the week number seemed to correspond to my expectations (though I'm not an expert on such things).  YMMV

**Comment by trainbuffer** [comment]
Obsidian v0.15.9
macOS v12.5
Periodic Notes v 0.0.17
Calendar v 1.5.10

I seem to be having this issue as well. I've set periodic notes to use `GGGG-[W]WW` for the format. It displays 2022-W33 as my syntax (which is correct as of me writing this). But if I create a new weekly note, the week is wrong. 
{{title}} outputs: 2022-W32
{{date:[Week] ww}} outputs: Week 34 

Not sure what's going on. My Calendar plugin is set to start the week on Sunday; and the locale is same as system. I'll try changing to "en-gb" and report back.

**Comment by doomgutt** [comment]
Obsidian v0.15.9
Ubuntu 22.04.1
Periodic Notes v 0.0.17

Same here, I can get it to make a note for one week before or after the correct one, lol.

The correct week number for me is 34.
The format gggg-[W]ww shows up as W35 in the settings and makes a note titled  2022-W35.
The format gggg-[W]WW shows up as W34 in the settings (correct) and makes a note titled 2022-W33, and every consecutive press of the shortcut results in an "Unable to create new file" message (doesn't happen for gggg-[W]ww).

Thanks for the suggestions so far, I'm not sure how locales work on Linux, I'll mess about with it.

**Comment by linhtto** [comment]
This is exactly the same behavior I get on my MacOS. And changing the locale doesn't help at all.

**Comment by xurizaemon** [comment]
What TZ are folk experiencing this bug in? I'm in UTC+12 currently and I see the wrong week get calculated. (I ask because when I hit date behaviour bugs, I often find they are specific to "those of us who live in the near future".)

For me, steps to reproduce are:

- `templates/Daily Note.md` contains:
  - `day: {{date:YYYY-MM-DD}}`
  - `- Week: [{{date:GGGG-[W]WW}}]({{date:YYYY/[W]WW}}.md) of {{date:GGGG}}`
- `templates/Weekly Note.md` contains:
  - `week: {{date:GGGG-[W]WW}}`
- It is Wednesday 2022-09-07
- `date +%U` outputs `36`
- Creating a new daily note on Wednesday 2022-09-07 generates `2022/09/07.md` with content:
  - `day: 2022-09-07`
  - `- Week: [2022-W36](2022/W36.md) of 2022`
- Creating a new weekly note on Wednesday 2022-09-07 generates `2022/W37.md` with content:
  - `week: 2022-W37`

Using the date complication to move from day to week does this:

![today-last-week](https://user-images.githubusercontent.com/105608/188852594-ad7052ff-5711-4c02-8b3a-3a6fd0e81478.gif)

**Comment by linhtto** [comment]
^
UTC-04

**Comment by marclipovsky** [comment]
The issue I'm seeing is that only creating a weekly note does this and not daily. This tells me the that logic to create a daily note is using the locale instead of ISO and daily note creation is using ISO. The preview in the settings also is using ISO.

![CleanShot 2022-09-20 at 11 47 57](https://user-images.githubusercontent.com/112484/191305344-ea6f2096-2464-4a97-bfa6-1ada6faf312b.png)

![CleanShot 2022-09-20 at 11 46 54](https://user-images.githubusercontent.com/112484/191305371-aa58608a-eaba-4db8-b6f6-0bc399f2239b.png)

**Comment by xurizaemon** [comment]
https://github.com/liamcain/obsidian-periodic-notes/issues/133#issue-1341647352 looks like a duplicate?

**Comment by marclipovsky** [comment]
There may be issues of Calendar + Periodic Notes plugin, but the issues described here are unrelated to it.

**Comment by linhtto** [comment]
I agree that it's only an issue with the Weekly Note, and it seems entirely fixable because it's correct elsewhere (in the Setting Preview and also as stated above, in the Daily Note)

**Comment by alexlyzhov** [comment]
Setting the locale to `en-gb` helped. In particular, install the Calendar plugin, set it to `en-gb` in plugin settings, optionally set Sunday as start of the week (again, in Calendar plugin settings) and reboot Obsidian.

**Comment by Julien-Gio** [comment]
I don't know the code base, but looks like PR #141 fixes the issue no?
